src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java

```diff
@@ -237,6 +240,33 @@ public Builder sdkMetricsTags(Map<String, String> tags) {
       return this;
     }
 
+  /**
+   * For a given server endpoint, validate according to RFC 2396 and attempt
+   * to make a connection
+   *
+   * @return {@code this}
+   * @throws IllegalStateException
+   */
+    public Builder validateEndpoint() throws IllegalStateException {
+      URL url = null;
+
+      try {
+        url = new URL(this.server);
+      } catch (MalformedURLException e) {
+        throw new IllegalStateException("Please provide a reachable Wavefront endpoint", e);
```

Comment:
semantically, IllegalArgumentException fits better here; and error message should be less confusing - something like `this.server + " is not a valid URL"` would work better

Reply:
I agree. This exception was requested, but I can change it.